tests/test_it.py

```diff
@@ -140,6 +148,16 @@ def test_C402_fail_2():
     ]
 
 
+def test_C402_fail_3():
+    errors = run_flake8("""
+        lst = [('a', 1), ('b', 2), ('c', 3)]
+        dict(pair for pair in lst if pair[1] % 2 == 0)
```

Comment:
The particular case can be rewritten yes, but it can't be automatically detected without knowing that `lst` is a list of two-tuples, which is not information that can be found anywhere in the AST (in general), as Python isn't statically typed